src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,262 @@
+import TableBody from "@vertigis/web/ui/TableBody";
+import TableCell, { TableCellProps } from "@vertigis/web/ui/TableCell";
+import TableFooter from "@vertigis/web/ui/TableFooter";
+import TableHead from "@vertigis/web/ui/TableHead";
+import TableRow from "@vertigis/web/ui/TableRow";
+
+type FooterRow = Record<string, any>;
+type Row = Record<string, any>;
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    alias?: string;
+    align?: TableCellProps["align"];
+    format?: (value: any) => any;
+    parse?: (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => any;
+    props?: any;
+    columnCalculation?: "sum" | "average" | ((values: any) => any);
+    editable: boolean;
+    type: "text" | "number" | "date" | "time" | string;
+    name: string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<Row[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: Row[];
+    editTooltip: string;
+    saveTooltip: string;
+    footerLabel?: string;
+    onMouseEnter?: (row: Row) => void;
+    onMouseLeave?: (row: Row) => void;
+    rowCalculation?: (data: Row[], row: Row) => void;
+}
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    lineHeight: "2.4rem",
+    display: "table-cell",
+    verticalAlign: "inherit",
+    textAlign: "left",
+    overflowWrap: "break-word",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    fontSize: "1.4rem",
+    color: "var(--secondaryForeground)",
+};
+
+const calculateFooterRow = (cols: Column[], rows: Row[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
```

Comment:
A table form element that supports inline editing

or something to that effect

Reply:
updated